crates/types/src/hotshot_config_file.rs

```diff
@@ -0,0 +1,158 @@
+    pub next_view_timeout: u64,
+    /// Duration for view sync round timeout
+    pub view_sync_timeout: Duration,
+    /// The exponential backoff ration for the next-view timeout
+    pub timeout_ratio: (u64, u64),
+    /// The delay a leader inserts before starting pre-commit, in milliseconds
+    pub round_start_delay: u64,
+    /// Delay after init before starting consensus, in milliseconds
+    pub start_delay: u64,
+    /// Number of network bootstrap nodes
+    pub num_bootstrap: usize,
+    /// The maximum amount of time a leader can wait to get a block from a builder
+    pub builder_timeout: Duration,
+    /// Time to wait until we request data associated with a proposal
+    pub data_request_delay: Option<Duration>,
+    /// Builder API base URL
+    #[serde(default = "default_builder_urls")]
+    pub builder_urls: Vec1<Url>,
+    /// Upgrade config
+    pub upgrade: UpgradeConfig,
+}
+
+impl<KEY: SignatureKey> From<HotShotConfigFile<KEY>> for HotShotConfig<KEY> {
+    fn from(val: HotShotConfigFile<KEY>) -> Self {
+        HotShotConfig {
+            execution_type: ExecutionType::Continuous,
+            start_threshold: val.start_threshold,
+            num_nodes_with_stake: val.num_nodes_with_stake,
+            known_da_nodes: val.known_da_nodes,
+            known_nodes_with_stake: val.known_nodes_with_stake,
+            known_nodes_without_stake: val.known_nodes_without_stake,
+            my_own_validator_config: val.my_own_validator_config,
+            da_staked_committee_size: val.staked_da_nodes,
+            fixed_leader_for_gpuvid: val.fixed_leader_for_gpuvid,
+            next_view_timeout: val.next_view_timeout,
+            view_sync_timeout: val.view_sync_timeout,
+            timeout_ratio: val.timeout_ratio,
+            round_start_delay: val.round_start_delay,
+            start_delay: val.start_delay,
+            num_bootstrap: val.num_bootstrap,
+            builder_timeout: val.builder_timeout,
+            data_request_delay: val
+                .data_request_delay
+                .unwrap_or(Duration::from_millis(REQUEST_DATA_DELAY)),
+            builder_urls: val.builder_urls,
+            start_proposing_view: val.upgrade.start_proposing_view,
+            stop_proposing_view: val.upgrade.stop_proposing_view,
+            start_voting_view: val.upgrade.start_voting_view,
+            stop_voting_view: val.upgrade.stop_voting_view,
+            start_proposing_time: val.upgrade.start_proposing_time,
+            stop_proposing_time: val.upgrade.stop_proposing_time,
+            start_voting_time: val.upgrade.start_voting_time,
+            stop_voting_time: val.upgrade.stop_voting_time,
+        }
+    }
+}
+
+impl<KEY: SignatureKey> From<ValidatorConfigFile> for HotShotConfig<KEY> {
+    fn from(value: ValidatorConfigFile) -> Self {
+        let mut config: HotShotConfig<KEY> = HotShotConfigFile::default().into();
+        config.my_own_validator_config = value.into();
+        config
+    }
+}
```

Comment:
I'm uneasy about this because I feel like `From` is for conversions that are "obvious" (whereas this is defaulting the HotShotConfig to a completely arbitrary value)

Reply:
This isn't even used. Nuked.